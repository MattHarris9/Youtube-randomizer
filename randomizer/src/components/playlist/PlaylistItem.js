import React from 'react';
import FancyPlaylistItem from './FancyPlaylistItem';
import SimplePlaylistItem from './SimplePlaylistItem';

const PlaylistItem = ({
  id,
  title,
  thumbUrl,
  onClick,
  visualEffectsEnabled,
  onDelete,
  onRefresh,
}) =>
  visualEffectsEnabled ? (
    <FancyPlaylistItem
      onClick={onClick}
      id={id}
      title={title}
      thumbUrl={thumbUrl}
      onDelete={onDelete}
      onRefresh={onRefresh}
    />
  ) : (
    <SimplePlaylistItem
      onClick={onClick}
      id={id}
      title={title}
      onDelete={onDelete}
      onRefresh={onRefresh}
    />
  );

export default React.memo(PlaylistItem);
