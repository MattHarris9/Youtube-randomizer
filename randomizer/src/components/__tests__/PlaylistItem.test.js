import React from 'react';
import renderer from 'react-test-renderer';
import PlaylistItem from '../playlist/PlaylistItem';

const mockFn = jest.fn();

describe('Playlist Item', () => {
  test('renders empty item correctly', () => {
    const tree = renderer.create(<PlaylistItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders full item correctly', () => {
    const props = {
      id: 1,
      title: 'Playlist #1',
      thumbUrl: 'https://i.ytimg.com/vi/0iMell14sVQ/default.jpg',
      onClick: mockFn,
    };
    const tree = renderer.create(<PlaylistItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
