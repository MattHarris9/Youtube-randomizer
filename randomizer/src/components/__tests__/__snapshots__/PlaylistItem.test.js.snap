// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Item renders empty item correctly 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      />
    </div>
    <div
      className="MuiListItemText-root"
    />
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;

exports[`Playlist Item renders full item correctly 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
    id={1}
    onClick={[MockFunction]}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle"
      >
        <img
          className="MuiAvatar-img"
          src="https://i.ytimg.com/vi/0iMell14sVQ/default.jpg"
        />
      </div>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Playlist #1
      </span>
    </div>
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;
